src/WebJobs.Script.WebHost/Extensions/FunctionMetadataExtensions.cs

```diff
@@ -76,25 +76,36 @@ public static async Task<FunctionMetadataResponse> ToFunctionMetadataResponse(th
         /// </summary>
         /// <param name="functionMetadata">FunctionMetadata object to convert to a JObject.</param>
         /// <param name="config">ScriptHostConfiguration to read RootScriptPath from.</param>
-        /// <returns>JObject that represent the trigger for scale controller to consume</returns>
-        public static async Task<JObject> ToFunctionTrigger(this FunctionMetadata functionMetadata, ScriptJobHostOptions config)
+        /// <param name="isFlexConsumption">True if this is for flex consumption, false otherwise.</param>
+        /// <returns>JObject that represent the trigger for scale controller to consume.</returns>
+        public static async Task<JObject> ToFunctionTrigger(
+            this FunctionMetadata functionMetadata, ScriptJobHostOptions config, bool isFlexConsumption = false)
         {
             var functionPath = Path.Combine(config.RootScriptPath, functionMetadata.Name);
             var functionMetadataFilePath = Path.Combine(functionPath, ScriptConstants.FunctionMetadataFileName);
 
             // Read function.json as a JObject
             var functionConfig = await GetFunctionConfig(functionMetadata, functionMetadataFilePath);
 
-            if (functionConfig.TryGetValue("bindings", out JToken value) &&
-                value is JArray)
+            if (functionConfig.TryGetValue("bindings", out JToken value) && value is JArray jArray)
             {
                 // Find the trigger and add functionName to it
-                foreach (JObject binding in (JArray)value)
+                foreach (JObject binding in jArray)
                 {
                     var type = (string)binding["type"];
                     if (type != null && type.EndsWith("Trigger", StringComparison.OrdinalIgnoreCase))
                     {
                         binding.Add("functionName", functionMetadata.Name);
+
+                        if (isFlexConsumption)
+                        {
+                            string group = functionMetadata.GetFunctionGroup();
+                            if (!string.IsNullOrEmpty(group))
+                            {
+                                binding.Add("functionGroup", group);
```

Comment:
What would be the best way to grab this value for tags without having to make the metrics meter be aware of function metadata

Reply:
My next PR will work on that aspect. Spoiler: it is just `FUNCTIONS_TARGET_GROUP` from `IEnvironment`